debezium-connector-oracle/src/test/java/io/debezium/connector/oracle/OracleConnectorIT.java

```diff
@@ -947,6 +948,220 @@ public void shouldPropagateSourceTypeByDatatype() throws Exception {
+                    + "values (3, INTERVAL '2' YEAR, INTERVAL '555-4' YEAR(3) TO MONTH, "
+                    + "INTERVAL '3' DAY, INTERVAL '111 10:09:08.555444333' DAY(3) TO SECOND(9))");
+            connection.execute("INSERT INTO debezium.interval (id, intYM, intYM2, intDS, intDS2) "
+                    + "values (4, INTERVAL '0' YEAR, INTERVAL '0' MONTH, "
+                    + "INTERVAL '0' DAY, INTERVAL '0' SECOND)");
+            connection.commit();
+
+            // Verify record generated during streaming
+            List<SourceRecord> streamingRecords = consumeRecordsByTopic(2).recordsForTopic("server1.DEBEZIUM.INTERVAL");
+            assertThat(streamingRecords.size()).isEqualTo(2);
+
+            streamingRecords.forEach(rec -> {
+                Struct after = ((Struct) rec.value()).getStruct(AFTER);
+                Struct key = (Struct) rec.key();
+                switch ((int) key.get("ID")) {
+                    case 3:
+                        assertThat(after.getInt64("INTYM")).isEqualTo(63115200000000L);
+                        assertThat(after.getInt64("INTYM2")).isEqualTo(17524987200000000L);
+                        assertThat(after.getInt64("INTDS")).isEqualTo(259200000000L);
+                        assertThat(after.getInt64("INTDS2")).isEqualTo(9627503444333L);
+                        break;
+                    case 4:
+                        assertThat(after.getInt64("INTYM")).isEqualTo(0L);
+                        assertThat(after.getInt64("INTYM2")).isEqualTo(0L);
+                        assertThat(after.getInt64("INTDS")).isEqualTo(0L);
+                        assertThat(after.getInt64("INTDS2")).isEqualTo(0L);
+                        break;
+                    default:
+                        fail("unexpected id");
+                }
+            });
+        }
+        finally {
+            TestHelper.dropTable(connection, "debezium.interval");
+        }
+    }
+
+    @Test
+    @FixFor("DBZ-1539")
+    public void shouldHandlerIntervalTypesAsString() throws Exception {
+        // Drop table if it exists
+        TestHelper.dropTable(connection, "debezium.interval");
+
+        try {
+            // complex ddl
+            final String ddl = "create table debezium.interval (" +
+                    " id numeric(6) constraint interval_id_nn not null, " +
+                    " intYM interval year to month," +
+                    " intYM2 interval year(9) to month," + // max precision
+                    " intDS interval day to second, " +
+                    " intDS2 interval day(9) to second(9), " + // max precision
+                    " constraint interval_pk primary key(id)" +
+                    ")";
+
+            // create table
+            connection.execute(ddl);
+            connection.execute("GRANT SELECT ON debezium.interval to " + TestHelper.getConnectorUserName());
+            connection.execute("ALTER TABLE debezium.interval ADD SUPPLEMENTAL LOG DATA (ALL) COLUMNS");
```

Comment:
Just one recommendation here in the future, we do have a `TestHelper#streamTable` method that should be used for these calls going forward as it does both of these steps.  This is just helpful that if we find we need to adjust or add to this behavior in the future that changes to enable streaming a table can be made in one place rather than per-test.

Reply:
makes a lot of sense. Copy/pasted this from the test above :-)